proxy/coco/src/peer/run_state.rs

```diff
@@ -42,44 +57,83 @@ pub enum Command {
     StartSyncTimeout(Duration),
 }
 
-impl From<RequestCommand> for Command {
-    fn from(other: RequestCommand) -> Self {
-        Self::Request(other)
-    }
+/// Reactions for incoming control requests.
+#[derive(Debug)]
+pub enum ControlCommand {
+    /// Send a response corresponding to a control request.
+    Respond(control::Response),
 }
 
 /// Commands issued when requesting an identity from the network.
 #[derive(Debug, PartialEq)]
 pub enum RequestCommand {
     /// Tell the subroutine to attempt a clone from the given `RadUrl`.
     Clone(RadUrl),
-    /// Tell the subroutine that the given `RadUrl` was found on the network.
-    Found(RadUrl),
     /// Tell the subroutine that we should query for the given `RadUrn` on the network.
     Query(RadUrn),
 }
 
-/// Significant events that occur during [`Peer`] lifetime.
+impl From<RequestCommand> for Command {
+    fn from(other: RequestCommand) -> Self {
+        Self::Request(other)
+    }
+}
+
+/// Events external subscribers can observe for internal peer operations.
 #[allow(clippy::large_enum_variant)]
 #[derive(Clone, Debug)]
 pub enum Event {
-    /// Ping the `select!` event loop periodically.
-    Ping,
+    /// Announcement subroutine completed and emitted the enclosed updates.
+    Announced(announcement::Updates),
+    /// An event from the underlying coco network stack.
+    Protocol(ProtocolEvent<Gossip>),
+    /// Sync with a peer completed.
+    PeerSynced(PeerId),
+    /// Request fullfilled with a successful clone.
+    RequestCloned(RadUrl),
+    /// Requested urn was queried on the network.
+    RequestQueried(RadUrn),
+    /// Waiting room interval ticked.
+    RequestTick,
+}
+
+impl MaybeFrom<&Input> for Event {
```

Comment:
Are we using `MaybeFrom` in any other places? _Maybe_ we just need a function :shrug:

Reply:
Went down that path, but it feels an abuse. It's not the same semantics. I can drop the `trait` if we feel strongly about it.

Reply:
I’d strongly favor removing this abstraction but will leave it up to you.

Reply:
I'd second that because it's only used in one place. If you wanted to abstract we could do our favourite impl trick!
```
impl From<&Input> for Option<Event>`
```

Reply:
Oh wait, no you can't

Reply:
@FintanH I tried that, dunno how to call `from` correctly with that impl.

Reply:
Ya, I was wrong anyway because we can't impl `From`/Into` for `Option`